halogen/vnd/error.py

```diff
@@ -0,0 +1,72 @@
+"""vnd.error support."""
+
+import halogen
+import six
+
+
+class Error(Exception):
+    """Base exception."""
+
+    def __init__(self, message, path=None, errors=None):
+        """Create an error.
+
+        :param message: Error message.
+        :param path: Optional JSON Pointer path.
+        :param errors: Optional nested errors.
+        """
+        self.message = message
+        if path is not None:
+            self.path = path
+        if errors is not None:
+            self.errors = errors
+
+    @classmethod
+    def from_validation_exception(cls, exception, **kwargs):
+        """Create an error from validation exception."""
+        errors = []
+
+        def flatten(error, path=""):
+            if isinstance(error, halogen.exceptions.ValidationError):
+                if not path.endswith("/"):
+                    path += "/"
+                if error.attr is not None:
+                    path += error.attr
+                elif error.index is not None:
+                    path += str(error.index)
```

Comment:
```suggestion
                    path += six.text_type(error.index)
```

Reply:
I don't know.. These are the keys of json or python attributes. I don't think i want to go into unicode story there.

Reply:
there is no unicode story, everything should be unicode, unless you are dealing with raw bytes. `path` should also default to the unicode empty string `u''`